# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerDebuggerManagerParent.cpp
# Commit: 9e22fbf8ab8f
# Full Hash: 9e22fbf8ab8f828b6e500a6e4994accc42c5e819
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-04-09 04:12:49
# Regressor Bug: 1899503
# File Overlap Count: 2
# Description:
#   Bug 1899503 - P1 New top-level IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger to support nsIWorkerDebugger interface for Workers in the parent process. r=asuth
#   
#   New IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger are introduced to support remote debugger of Workers on the parent thread.
#   
#   PRemoteWorkerDebuggerManager is IPC between the parent process main thread and the content process WorkerLauncher thread.
# ==============================================================================

diff -r a5880ebdc3d7 -r 9e22fbf8ab8f dom/workers/remoteworkers/RemoteWorkerDebuggerManagerParent.cpp
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerDebuggerManagerParent.cpp	Tue Apr 08 09:43:33 2025 +0000
@@ -0,0 +1,38 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "RemoteWorkerDebuggerManagerParent.h"
+#include "RemoteWorkerDebuggerParent.h"
+#include "mozilla/dom/WorkerDebuggerManager.h"
+
+namespace mozilla::dom {
+
+RemoteWorkerDebuggerManagerParent::RemoteWorkerDebuggerManagerParent() {
+  MOZ_ASSERT_DEBUG_OR_FUZZING(XRE_IsParentProcess() && NS_IsMainThread());
+}
+
+RemoteWorkerDebuggerManagerParent::~RemoteWorkerDebuggerManagerParent() {
+  MOZ_ASSERT_DEBUG_OR_FUZZING(XRE_IsParentProcess() && NS_IsMainThread());
+}
+
+mozilla::ipc::IPCResult RemoteWorkerDebuggerManagerParent::RecvRegister(
+    const RemoteWorkerDebuggerInfo& aDebuggerInfo,
+    mozilla::ipc::Endpoint<PRemoteWorkerDebuggerParent>&& aParentEp) {
+  MOZ_ASSERT_DEBUG_OR_FUZZING(XRE_IsParentProcess() && NS_IsMainThread());
+  RefPtr<WorkerDebuggerManager> manager = WorkerDebuggerManager::Get();
+  MOZ_ASSERT_DEBUG_OR_FUZZING(manager);
+
+  nsCOMPtr<nsIWorkerDebugger> debugger =
+      manager->GetDebuggerById(aDebuggerInfo.Id());
+  MOZ_ASSERT_DEBUG_OR_FUZZING(!debugger);
+
+  debugger = MakeRefPtr<RemoteWorkerDebuggerParent>(aDebuggerInfo,
+                                                    std::move(aParentEp));
+
+  manager->RegisterDebugger(debugger);
+
+  return IPC_OK();
+}
+
+}  // end of namespace mozilla::dom